runtime/doc/pack.txt

```diff
@@ -355,16 +356,27 @@ del({names})                                                  *vim.pack.del()*
                  be managed by |vim.pack|, not necessarily already added to
                  current session.
 
-get()                                                         *vim.pack.get()*
-    Get data about all plugins managed by |vim.pack|
+get({names}, {opts})                                          *vim.pack.get()*
+    Get data about plugins managed by |vim.pack|
+
+    Parameters: ~
+      • {names}  (`string[]?`) List of plugin names. Default: all plugins
+                 managed by |vim.pack|.
+      • {opts}   (`table?`) A table with the following fields:
+                 • {details} (`boolean`) Whether to include plugin details.
```

Comment:
Can be 3 flags `branch`, `tags`, etc. That avoids the urge to put that info into a dict. Or name it `scm` or `vcs`, and it controls VCS-related fields (but they are not nested).

Reply:
On one hand, this can allow more precise control over performance (only enable flags which you want to see).

On the other hand, won't it become more confusing if the number of provided "extra" details grow? For example, there will probably also be `type` field if/when there are "local plugins". Plus the reasonable set of useful flags might be different for different plugin types.

Weighing both options, I'd probably still go with a single `details` / `info` flag that populates all extra available data about the plugin. For Git plugins they are computed in parallel, so shouldn't be too much of a problem. But I'd go the way you suggest eventually, of course.

Reply:
> True. In #30086 we are considering (the opposite concept) `fast`. And git has the concept of `sparse` checkouts.

I mean, it can be `fast` also. This is mostly the main reason of the flag's existence anyways: allow a way to control whether there needs to be time/resources spent to gather more information. Useful for both users and `vim.pack` itself (the `fast = true` or `info = false` case).

If `fast` is in consideration, then I'd go with it.

Reply:
idk if `fast` is a good pattern to proliferate, because (1) performance is variable, (2) it has lower semantic value

let's say `info` for now